apps/dashboard/src/components/workflow-editor/configure-workflow-form.tsx

```diff
@@ -20,61 +20,56 @@ import { Switch } from '../primitives/switch';
 import { TagInput } from '../primitives/tag-input';
 import { Textarea } from '../primitives/textarea';
 import { MAX_DESCRIPTION_LENGTH, workflowSchema } from '@/components/workflow-editor/schema';
-import { z } from 'zod';
+import { useFormAutosave } from '@/hooks/use-form-autosave';
 
 type ConfigureWorkflowFormProps = {
   workflow: WorkflowResponseDto;
-  debouncedUpdate: (data: UpdateWorkflowDto) => void;
-  patch: (data: PatchWorkflowDto) => void;
+  update: (data: UpdateWorkflowDto) => void;
   onDirtyChange: (isDirty: boolean) => void;
 };
 
 export const ConfigureWorkflowForm = (props: ConfigureWorkflowFormProps) => {
-  const { workflow, debouncedUpdate, patch, onDirtyChange } = props;
+  const { workflow, update, onDirtyChange } = props;
   const isReadOnly = workflow.origin === WorkflowOriginEnum.EXTERNAL;
   const [isPauseModalOpen, setIsPauseModalOpen] = useState(false);
   const tagsQuery = useTagsQuery();
 
   const form = useForm<z.infer<typeof workflowSchema>>({
     defaultValues: {
+      active: workflow.active,
       name: workflow.name,
       workflowId: workflow.workflowId,
       description: workflow.description,
       tags: workflow.tags,
     },
     resolver: zodResolver(workflowSchema),
+    shouldFocusError: false,
   });
 
-  const statusForm = useForm({
-    defaultValues: {
-      active: workflow.active,
-    },
+  const { onBlur, flushFormUpdates } = useFormAutosave({
+    previousData: workflow,
+    form,
+    isReadOnly,
+    update,
```

Comment:
The modified `useFormAutosave` that provides the form `onBlur` listener and the utility function to flush all form changes and call api. More about it later.